src/OpenTelemetry.Shims.OpenTracing/SpanShim.cs

```diff
@@ -183,7 +188,7 @@ public string GetBaggageItem(string key)
             }
             else
             {
-                this.Span.SetAttribute(key, value);
+                this.Span.SetAttribute(key, value.ToString());
```

Comment:
we'll be able to avoid this conversions once new changes in Activity arrives in preview 8.